src/vs/workbench/contrib/webview/browser/webviewView.ts

```diff
@@ -0,0 +1,199 @@
+		@IWebviewService private readonly webviewService: IWebviewService,
+	) {
+		super({ ...options, titleMenuId: MenuId.ViewTitle }, keybindingService, contextMenuService, configurationService, contextKeyService, viewDescriptorService, instantiationService, openerService, themeService, telemetryService);
+
+		this._register(this.onDidChangeBodyVisibility(() => this.updateTreeVisibility()));
+		this.updateTreeVisibility();
+	}
+
+	focus(): void {
+		super.focus();
+		this._webview?.focus();
+	}
+
+	renderBody(container: HTMLElement): void {
+		super.renderBody(container);
+
+		this._container = container;
+
+		if (!this._resizeObserver) {
+			this._resizeObserver = new ResizeObserver(() => {
+				setImmediate(() => {
+					if (this._container) {
+						this._webview?.layoutWebviewOverElement(this._container);
+					}
+				});
+			});
+
+			this._register(toDisposable(() => {
+				this._resizeObserver.disconnect();
+			}));
+			this._resizeObserver.observe(container);
+		}
+	}
+
+	protected layoutBody(height: number, width: number): void {
+		super.layoutBody(height, width);
+
+		if (!this._webview) {
+			return;
+		}
+
+		if (this._container) {
+			this._webview.layoutWebviewOverElement(this._container, { width, height });
+		}
+	}
+
+	private updateTreeVisibility() {
+		if (this.isBodyVisible()) {
+			this.activate();
+			this._webview?.claim(this);
+		} else {
+			this._webview?.release(this);
+		}
+	}
+
+	private activate() {
+		if (!this._activated) {
+			this._activated = true;
+
+			const webview = this.webviewService.createWebviewOverlay(generateUuid(), {}, {}, undefined);
+			this._webview = webview;
+
+			this._register(toDisposable(() => {
+				this._webview?.release(this);
+			}));
+
+			this.extensionService.activateByEvent(`onView:${this.id}`);
+
+			this.withProgress(() => {
+				return Registry.as<IViewWebviewViewResolverRegistry>(Extensions.WebviewViewResolverRegistry)
+					.resolve(this.id, webview);
+			});
+		}
+	}
+
+	private async withProgress(task: () => Promise<void>): Promise<void> {
+		return this.progressService.withProgress({ location: this.id, delay: 500 }, task);
+	}
+}
+
+const viewResolverRegistry = Registry.as<IViewResolverRegistry>(ViewsExtensions.ViewResolverRegistry);
+viewResolverRegistry.register('webview', {
+	resolve: (viewDescriptor: IViewDescriptor): IViewDescriptor => {
+		return {
+			...viewDescriptor,
+			ctorDescriptor: new SyncDescriptor(WebviewViewPane)
+		};
+	}
+});
+
+
+export namespace Extensions {
+	export const WebviewViewResolverRegistry = 'workbench.registry.webviewViewResolvers';
+}
+
+export interface IWebviewViewResolver {
+	resolve(webview: Webview): Promise<void>;
+}
+
+export interface IViewWebviewViewResolverRegistry {
+	register(type: string, resolver: IWebviewViewResolver): IDisposable;
+
+	resolve(viewType: string, webview: Webview): Promise<void>;
+}
+
+class WebviewViewResolverRegistry extends Disposable implements IViewWebviewViewResolverRegistry {
```

Comment:
I think services are the better singletons and unless this really needs to be static, e.g be used while loading code, I would try to put this into some service.